### PR TITLE
Added a script to check the installed node version

### PR DESCRIPTION
**Pull request by Raibaz**
This should solve #82 with [this](http://stackoverflow.com/questions/29349684/how-can-i-specify-the-required-node-js-version-in-packages-json/41620850#41620850) solution proposed in the issue comments.

A script has been added in the `scripts` folder to check the current node version against the one required in `package.json` and the script execution has been bound to the `preinstall` stage in the package file itself.

Unfortunately this adds a dependency from `semver`, but I guess there is no easier way to handle version comparison than adding an external dependency.

### CODE REVIEW

**Comment by Raibaz** [comment]
Note that i had to add `"test": "mocha --recursive"` to explicitly define the test suite runner script, since having a `scripts` property in `package.json` overrides the defaults.

File: scripts/check-version.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const semver = require('semver');
+const engines = require('../package').engines;
+
+const version = engines.node;
+if (!semver.satisfies(process.version, version)) {
+  console.log(`Required node version ${version} not satisfied with current version ${process.version}.`);
+  process.exit(1);
+} else {
```
Comment (evykassirer) [inline scripts/check-version.js]: I added this to #44, but can you put the else on a new line?

````  process.exit(1)
}
else {

File: scripts/check-version.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const semver = require('semver');
+const engines = require('../package').engines;
+
+const version = engines.node;
+if (!semver.satisfies(process.version, version)) {
+  console.log(`Required node version ${version} not satisfied with current version ${process.version}.`);
```
Comment (evykassirer) [inline scripts/check-version.js]: 80 char max per line please :)

File: package.json [hunk 3]
```diff
@@ -34,5 +38,10 @@
   "bugs": {
     "url": "https://github.com/socraticorg/mathsteps/issues"
   },
-  "homepage": "https://github.com/socraticorg/mathsteps#readme"
+  "homepage": "https://github.com/socraticorg/mathsteps#readme",
+  "scripts": {
+    "requirements-check": "node scripts/check-version.js",
+    "preinstall": "npm run requirements-check",
+    "test": "mocha --recursive"
```
Comment (evykassirer) [inline package.json]: you don't actually need to add this line :)

just add the ` "requirements-check"` and `"preinstall"` to the scripts starting on line 18

File: scripts/check-version.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const semver = require('semver');
+const engines = require('../package').engines;
+
+const version = engines.node;
+if (!semver.satisfies(process.version, version)) {
```
Comment (evykassirer) [inline scripts/check-version.js]: run `npm lint` and you'll see there are some errors

put `/*global process*/` at the top of this file to get rid of one

and then `// eslint-disable-next-line` above the `console.log` lines

File: scripts/check-version.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const semver = require('semver');
```
Comment (evykassirer) [inline scripts/check-version.js]: put `'use strict';` at the top of this file

**Comment by Raibaz** [comment]
Whops i didn't notice `package.json` already had a scripts section and I copypasted the stackoverflow solution without linting it...sorry :(

Should be ok now.

**Comment by evykassirer** [comment]
sweet! thanks for the quick fix

**Pull request merged**